### DISCUSSION TITLE
Is it possible to prevent PRs of a package version that exceed the version of another package?

### DISCUSSION THREAD

**Discussion by kieran-ryan** [renovatebot/renovate#26439] [Request Help] [answered]
### How are you running Renovate?

GitHub App

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Is it possible to configure a package ('A') to be a dependency of another ('B'), such that:

- PRs are not raised for 'A' if the version would exceed 'B'
- 'B' will not automerge
- 'A' update will be grouped into 'B' update - if both available
- 'A' will automerge if 'B' if a PR is raised for 'A' on its own e.g. in scenario where 'B' PR was raised and merged but associated 'A' release was not available

### Example

In the `package.json` of a VSCode extension, you can set the VSCode compatibility version (`engine`). You also may want to specify the VSCode types you are using, which cannot exceed the engine?

```json
  "engines": {
    "vscode": "^1.82.0"
  }
  "devDependencies": {
    "@types/vscode": "1.82.0"
  }
```

Is there a way to prevent Renovate raising PRs for new versions of `@types/vscode` that exceed the minimum `vscode` version (1.82.0)? Group PRs where both can be updated? Automerge `@types/vscode` if associated version has already been raised for `vscode`.

Current `renovate.json`:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "github>cucumber/renovate-config"
  ],
  "packageRules": [
    {
      "matchPackageNames": ["vscode"],
      "automerge": false
    }
  ]
}
```

#### Rationale

`@types/vscode` will never be updated unless `vscode` is updated - so it can not be considered an open pull request unlike others. Reduce noise.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by viceice** [comment, marked as answer]
not possible
